straxen/rucio.py

```diff
@@ -268,44 +274,33 @@ def __init__(self, staging_dir, download_heavy=False, **kwargs):
             except OSError:
                 raise PermissionError(f"You told the rucio backend to download data to {staging_dir}, "
                                       f"but that path is not writable by your user")
-
         super().__init__(**kwargs)
         self.staging_dir = staging_dir
         self.download_heavy = download_heavy
-        # Do it only when we actually load rucio
-        from rucio.client.downloadclient import DownloadClient
-        self.download_client = DownloadClient()
-
-    def get_metadata(self, dset_did, rse='UC_OSG_USERDISK', **kwargs):
-        base_dir = os.path.join(self.staging_dir, did_to_dirname(dset_did))
-
-        # define where the metadata will go (or where it already might be)
-        number, dtype, hsh = parse_did(dset_did)
-        metadata_file = f"{dtype}-{hsh}-metadata.json"
-        metadata_path = os.path.join(base_dir, metadata_file)
-
-        # download if it doesn't exist
-        if not os.path.exists(metadata_path):
-            metadata_did = f'{dset_did}-metadata.json'
-            did_dict = dict(did=metadata_did,
-                            base_dir=base_dir,
-                            no_subdir=True,
-                            rse=rse
-                            )
-            print(f"Downloading {metadata_did}")
-            self._download([did_dict])
 
+    def get_metadata(self, dset_did, **kwargs):
+        if dset_did in self.dset_cache:
+            rse = self.dset_cache[dset_did]
+        else:
+            rses = admix.rucio.get_rses(dset_did)
+            rse = admix.downloader.determine_rse(rses)
+            self.dset_cache[dset_did] = rse
+
+        metadata_did = f'{dset_did}-metadata.json'
+        downloaded = admix.download(metadata_did, rse=rse, location=self.staging_dir)
+        assert len(downloaded) == 1, f"{metadata_did} should be a single file. We found {len(downloaded)}."
+        metadata_path = downloaded[0]
```

Comment:
while at it, do you see a way of fixing https://github.com/XENONnT/straxen/issues/586 and https://github.com/XENONnT/straxen/issues/670 which are caused by this piece of the code?

Reply:
Ok I think this should be semi-addressed now. The problem seemed to be with the other storage plugins. If I only loaded the rucio frontend it worked fine. I still don't fully understand the reason yet but I think it probably had to do with the fact that the rucio backend was downloading data be in same place that the other storage frontends were looking? 

Anyway after a few minor improvements and putting the rucio frontend to the front of the line,  this snippet now works for me: 
```
st = straxen.contexts.xenonnt(cmt_version='global_v3',
                              download_heavy=True
                             )
print(st.storage)
#st.storage = [st.storage[-1]]

run_id = '024345'
print(st.is_stored(run_id, 'raw_records'))

pc = st.get_array(run_id, 'raw_records', seconds_range=(0, 2))
print(pc.nbytes)
```

Reply:
Ah you are right, thanks a lot! 
The issue comes from the fact that we are using the same folder in two back ends with different frontends. Whereas the DataDirectory frontend considered something stored if both the folder is named correctly (since a complete write of the folder removes the _temp of the folder name), the logic for the RucioFrontend is different, it checks all the chunks exist.

I think the most robust solution is to just not re-use the exact same path in the two backends, would this be fine with you?